modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -1151,21 +1153,17 @@ private[codegen] class SmithyToIR(
       shape
         .members()
         .asScala
-        .map { member =>
-          val memberTarget =
-            model.expectShape(member.getTarget)
-          if (isPartOfAdt(memberTarget)) {
-            (member.getMemberName(), member.tpe.map(Left(_)))
-          } else {
-            (member.getMemberName(), member.tpe.map(Right(_)))
+        .flatMap { member =>
```

Comment:
note: this change is no longer necessary, it's merely a refactor. I left it in because the original seemed needlessly duplicative.